File: backend/src/main/java/com/xiaocui/followup/settings/SettingsController.java
package com.xiaocui.followup.settings;

import jakarta.validation.constraints.NotBlank;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@Validated
@RestController
@RequestMapping("/api/settings")
public class SettingsController {
    private final SettingsService service;

    public SettingsController(SettingsService service) {
        this.service = service;
    }

    @GetMapping("/ai")
    public SettingsService.AiSettingsView getAiSettings() {
        return service.view();
    }

    @PutMapping("/ai")
    public SettingsService.AiSettingsView saveAiSettings(@RequestBody AiSettingsRequest request) {
        service.save(
                request.enabled() != null && request.enabled(),
                request.baseUrl(),
                request.model(),
                request.apiKey()
        );
        return service.view();
    }

    @PostMapping("/ai/test")
    public TestResult testAiSettings(@RequestBody AiSettingsRequest request) {
        String reply = service.testConnection(true, request.baseUrl(), request.model(), request.apiKey());
        return new TestResult(true, reply);
    }

    public record AiSettingsRequest(Boolean enabled, String baseUrl, String model, String apiKey) {
    }

    public record TestResult(boolean ok, String reply) {
    }
}

File: backend-ts/src/controllers/settings.controller.ts
import { Body, Controller, Get, HttpCode, Post, Put } from '@nestjs/common';
import { SettingsService } from '../providers/settings.service';

/** AI 设置路由（等价原 Express 版 SettingsController 的路由块）。 */
@Controller('api/settings/ai')
export class SettingsController {
  constructor(private readonly settings: SettingsService) {}

  @Get()
  view() {
    return this.settings.view();
  }

  @Put()
  save(@Body() body: Record<string, any>) {
    const b = body ?? {};
    this.settings.save(b.enabled === true, b.baseUrl ?? null, b.model ?? null, b.apiKey ?? null);
    return this.settings.view();
  }

  @Post('test')
  @HttpCode(200)
  async test(@Body() body: Record<string, any>) {
    const b = body ?? {};
    const reply = await this.settings.testConnection(b.baseUrl ?? null, b.model ?? null, b.apiKey ?? null);
    return { ok: true, reply };
  }
}
